otlp/collector/go.mod

```diff
@@ -0,0 +1,35 @@
+module go.opentelemetry.io/proto/otlp/collector
+
+go 1.21.12
+
+replace go.opentelemetry.io/proto/otlp/common => ../common
+
+replace go.opentelemetry.io/proto/otlp/logs => ../logs
+
+replace go.opentelemetry.io/proto/otlp/metrics => ../metrics
+
+replace go.opentelemetry.io/proto/otlp/profiles => ../profiles
+
+replace go.opentelemetry.io/proto/otlp/resource => ../resource
+
+replace go.opentelemetry.io/proto/otlp/trace => ../trace
+
+require (
+	github.com/grpc-ecosystem/grpc-gateway/v2 v2.20.0
+	go.opentelemetry.io/proto/otlp/logs v0.0.0-00010101000000-000000000000
+	go.opentelemetry.io/proto/otlp/metrics v0.0.0-00010101000000-000000000000
+	go.opentelemetry.io/proto/otlp/profiles v0.0.0-00010101000000-000000000000
+	go.opentelemetry.io/proto/otlp/trace v0.0.0-00010101000000-000000000000
+	google.golang.org/grpc v1.65.0
```

Comment:
The OTLP HTTP exporters are depending on `go.opentelemetry.io/proto/otlp/collector`.
E.g. from `otlptracehttp`: https://github.com/open-telemetry/opentelemetry-go/blob/d4b5396f848504859ba50946e4724b84b71b36a5/exporters/otlp/otlptrace/otlptracehttp/client.go#L128
Therefore all actual OTLP exporters will still depend on `google.golang.org/grpc` and it will not help with solving https://github.com/open-telemetry/opentelemetry-go/issues/2579.

Reply:
I think the possible solutions are:

1. solving https://github.com/open-telemetry/opentelemetry-go/issues/2579#issuecomment-2071569890 in `google.golang.org/protobuf`
2. in `go.opentelemetry.io/proto/otlp/collecto` split protobuf messages and gRPC services code from into separate packages and modules

Reply:
PS. This was my first idea when I was trying to fix https://github.com/open-telemetry/opentelemetry-go/issues/2579. And I created `slim` because of https://github.com/open-telemetry/opentelemetry-proto-go/pull/183#discussion_r1689137778. At that point of time I was totally not aware of https://protobuf.dev/reference/go/faq/#namespace-conflict.

Reply:
You're right about the need for the collector even in HTTP, sorry.
I'll need to investigate more, but this PR, alongside  a new `collectorslim` module which only includes the collector slim, and reuses the other data types may solve our registration issue?

Reply:
> but this PR, alongside a new `collectorslim` module which only includes the collector slim, and reuses the other data types may solve our registration issue?

It could solve the issue. Probably this would require more post-processing after protoc code generation. Personally, I would be more in favor of checking if it is possible to solve somehow https://protobuf.dev/reference/go/faq/#namespace-conflict. Then we could simply use the `slim` modules in OTLP HTTP exporters. Moreover, it can also help others dealing with the namespace conflict.